ADR/nnnn-task-versioning.md

```diff
@@ -0,0 +1,361 @@
+# NN. Start versioning Tekton Tasks responsibly
+
+Date: 2025-10-06
+
+## Status
+
+Proposed
+
+Relates to [ADR NN. Trusted Tasks model after build-definitions decentralization](ADR/tbd.md).
```

Comment:
https://github.com/konflux-ci/architecture/pull/262

Reply:
The relation from the other direction is only tangential. Task versioning would be needed to make version-based rules work better, but apart from that the trusted task rules would work on their own

Reply:
On second thought, it doesn't hurt to have the link. But I think I'll add it after the PRs are merged